### PR TITLE
Upgrade apache http client to 4.5.8

### PR DESCRIPTION
**Pull request by mattdrees**
This is primarily to address CVE-2012-6153, but also CVE-2015-5262.

### CODE REVIEW

File: deployments-core/src/main/java/org/ccci/deployment/util/PageMatcher.java [hunk 1]
```diff
@@ -76,29 +75,24 @@ public Matcher pingUntilPageMatches(String uri, String regularExpression, String
...
-                        throw new RuntimeException(pageName + " not available; status line: " + statusLine);
+                    return matchPage(successPattern, regularExpression, pageName, content);
                 }
-                finally
-                //free up http connection
+                else if (statusLine.getStatusCode() == 404)
+                    log.debug("received 404; continuing to try to connect");
+                else if(statusLine.getStatusCode() == 400 && statusLine.getReasonPhrase().contains("No Host matches server name"))
                 {
-                    response.getEntity().consumeContent();
```
Comment (hlbraddock) [inline deployments-core/src/main/java/org/ccci/deployment/util/PageMatcher.java]: I take it `response.getEntity().consumeContent()` is no longer required due to the introduction of `final String content = EntityUtils.toString(response.getEntity())` (https://github.com/CruGlobal/deployments/pull/6/files#diff-96bc565aa35a123a166dd274313949c1R78).

Was that required as a result of the upgrade or simply a refactoring improvement?

File: deployments-ant-task/pom.xml [hunk 2]
```diff
@@ -1,14 +1,14 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
+
   <parent>
     <artifactId>deployments-parent</artifactId>
     <groupId>org.ccci</groupId>
-    <version>3-SNAPSHOT</version>
+    <version>3</version>
   </parent>
-  <groupId>org.ccci</groupId>
```
Comment (hlbraddock) [inline deployments-ant-task/pom.xml]: If you're cleaning up, you could also remove https://github.com/CruGlobal/deployments/blob/master/deployments-maven-plugin/pom.xml#L5 and https://github.com/CruGlobal/deployments/blob/master/deployments-command-line/pom.xml#L3.

**Review by hlbraddock — APPROVED** [review]

File: deployments-core/src/main/java/org/ccci/deployment/util/PageMatcher.java (hunk 1 above)
Comment (mattdrees) [reply deployments-core/src/main/java/org/ccci/deployment/util/PageMatcher.java]: Right.

Result of the upgrade.

File: deployments-ant-task/pom.xml (hunk 2 above)
Comment (mattdrees) [reply deployments-ant-task/pom.xml]: I did, actually. Those links point to master, not to my branch.

**Comment by mattdrees** [comment]
thanks!

File: deployments-ant-task/pom.xml (hunk 2 above)
Comment (hlbraddock) [reply deployments-ant-task/pom.xml:8]: Ahh, ok. I did know I sent you links to master, but I thought for some reason I'd seen them in your branch. My mistake. I'll just send links to the branch in future.

**Pull request merged**